Sources/StreamChat/Controllers/ChannelController/ChannelController.swift

```diff
@@ -147,7 +147,7 @@ public class ChatChannelController: DataController, DelegateCallable, DataStoreP
     /// It is `true` if the channel typing events are enabled as well as the user privacy settings.
     internal var shouldSendTypingEvents: Bool {
         /// Ignore if user typing indicators privacy settings are disabled. By default, they are enabled.
-        let currentUserPrivacySettings = client.currentUserController().currentUser?.privacySettings
+        let currentUserPrivacySettings = client.sharedCurrentUserController.currentUser?.privacySettings
```

Comment:
TBH, I don't think this is really needed 🤔  Having the `ChannelController` holding a `currentUserController` should be enough 🤔 WDYT?

Reply:
I considered this but then there is current user controller per channel controller (if user typed in that controller) which is not as efficient than limiting it to a single shared instance. Currently I went for more complex but more efficient solution. 
Slightly afraid that changes in muted channels triggers too many updates if multiple current user controllers are alive.

Reply:
I see, makes sense 👍 Lets do it like this then. Since the API is private, I'm good with both option 1 or 2 👍